blockchain/contracts/MainContract.sol

```diff
@@ -121,13 +123,13 @@ contract NFTtheWorld {
         }
         availableNFTsCount[dropHash] = availableNFTs[dropHash].length;
         numberOfDrops += 1;
-        reservedNFTsCount[dropHash] = 0;
     }
 
     // This function lets a user join a drop by specifying the number of NFTs she would like to reserve.
     function joinDrop(uint256 _numberOfNFTs, uint256 _dropHash) public {
         require(
-            reservedNFTsCount[_dropHash] != availableNFTs[_dropHash].length,
+            dropData[_dropHash].reservedCount !=
+                availableNFTs[_dropHash].length,
             "Cannot join the drop anymore."
```

Comment:
As now "availableNFTs" gets diminished (call to remove() ) within the function it doesnt represent the number of total NFTs available anymore, but the number of currently available NFTs. This throws error if at excactly 50% reserved NFTs of drop. Either:
- work with numberofURIs
- dont remove from availableNFTs at all